experimental/virtio/src/vsock/mod.rs

```diff
@@ -114,9 +118,35 @@ impl VSock {
         }
     }
 
+    /// Reads the next valid packet that matches the filter, if one is available.
+    ///
+    /// If `reset_unmatched` is true we send RST packets in response to unmatched packets to notify
+    /// the host that the pakcets are not part of a valid connected socket (e.g. unexpected source
+    /// or destintation ports).
+    pub fn read_filtered_packet<F: Fn(&Packet) -> bool>(
+        &mut self,
+        filter: F,
+        reset_unmatched: bool,
+    ) -> Option<Packet> {
+        loop {
```

Comment:
The loop here is only here in case there are multiple invalid packets in the queue to ensure they are all cleared out. If the loop isn't here the higher level code that reads data from a connected stream will not know whether the stream is currently empty or there was an invalid packet.